dask_ml/model_selection/_split.py

```diff
@@ -502,12 +559,44 @@ def train_test_split(
             blockwise=blockwise,
             random_state=random_state,
         )
-        train_idx, test_idx = next(splitter.split(*arrays))
 
-        train_test_pairs = (
-            (_blockwise_slice(arr, train_idx), _blockwise_slice(arr, test_idx))
-            for arr in arrays
-        )
+        if stratify is not None:
+            # convert to da.Array
+            if not isinstance(stratify, da.Array):
+                stratify = stratify.to_dask_array(lengths=True)
+
+            # must be 1-d for indexing
+            stratify = stratify.ravel()
+
+            train_test_pairs = []
+            for arr in arrays:
+                # list of class-wise train/test split array slices
+                arr_train_slices = []
+                arr_test_slices = []
+                for ci in classes:
+                    # get subarray of data from this class
+                    ci_arr = arr[stratify == ci]
+                    # FIXME: can chunks be determined lazily?
+                    ci_arr.compute_chunk_sizes()
```

Comment:
We'll want to rethink this a bit... `arrays` will likely have a common dependency structure (e.g. they may be read from disk at the same time). So we'll want to do as much as possible in the same call to compute. Right now it looks like we have a `compute` per array.

Reply:
Could an alternative be to force dask data-frames/series? This could alleviate the problem of the additional pass and avoid this additional `compute` per array